.github/workflows/X-Reusable-Test.yml

```diff
@@ -0,0 +1,134 @@
+name: X - Reusable Test Workflow 
+
+on:
+  workflow_call:
+    inputs:
+      jre:
+        description: 'The JRE that the tests will run on'
+        default: '8'
+        required: false
+        type: string
+    secrets:
+      aws-access-key-id:
+        required: true
+      aws-secret-access-key:
+        required: true
+
+jobs:
+  tests:
+    name: Java ${{ inputs.jre }} Instrumentation Tests
+    timeout-minutes: 60
+    # needs: install-all-java
+    runs-on: ubuntu-latest
+    env:
+      default-branch: "main"
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.BRANCH || env.default-branch }}
+      - uses: gradle/wrapper-validation-action@v1.0.4
+
+      #NEW Install required JDKs
+      # Install 8
+      - name: Set up Java 8
+        # https://github.com/actions/setup-java
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: 8
+          cache: 'gradle'
+
+      # Save new JDK variable
+      - name: Save JAVA_HOME as JDK8 for later usage
+        run: |
+          echo "Current JAVA_HOME = ${JAVA_HOME}"
+          echo "ORG_GRADLE_PROJECT_jdk8=$JAVA_HOME" >> $GITHUB_ENV
+
+      # Install 11
+      - name: Set up Java 11
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: 11
+
+      # Save new JDK variable
+      - name: Save JAVA_HOME as JDK11 for later usage
+        run: |
+          echo "Current JAVA_HOME = ${JAVA_HOME}"
+          echo "ORG_GRADLE_PROJECT_jdk11=$JAVA_HOME" >> $GITHUB_ENV
+
+      # Install 17
+      - name: Set up Java 17
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: 17
+
+      # Save new JDK variable
+      - name: Save JAVA_HOME as JDK17 for later usage
+        run: |
+          echo "Current JAVA_HOME = ${JAVA_HOME}"
+          echo "ORG_GRADLE_PROJECT_jdk17=$JAVA_HOME" >> $GITHUB_ENV
+
+      # Check ENV variables
+      - name: Check environmental variables
+        run: printenv | sort -f
+
+      ## AWS jars - plan to cache
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.aws-access-key-id }}
+          aws-secret-access-key: ${{ secrets.aws-secret-access-key }}
```

Comment:
Can't the reusable workflow access the repo secrets directly?
That would save a few lines in the concrete workflows.

Reply:
It failed when accessing directly from reusable.